Tests/SwiftSyntaxMacrosTest/MacroSystemTests.swift

```diff
@@ -726,6 +727,80 @@ public struct DeclsFromStringsMacroNoAttrs: DeclarationMacro {
   }
 }
 
+struct TracedPreambleMacro: PreambleMacro {
+  static func expansion(
+    of node: AttributeSyntax,
+    providingPreambleFor declaration: some DeclSyntaxProtocol & HasTrailingOptionalCodeBlock,
+    in context: some MacroExpansionContext
+  ) throws -> [CodeBlockItemSyntax] {
+    // FIXME: Should be able to support (de-)initializers and accessors as
+    // well, but this is a lazy implementation.
+    guard let funcDecl = declaration.as(FunctionDeclSyntax.self) else {
+      return []
+    }
+
+    let funcBaseName = funcDecl.identifier.text
+    let paramNames = funcDecl.signature.input.parameterList.map { param in
+      if let name = param.parameterName {
+        return name.text
+      } else {
+        return "_"
+      }
+    }
+
+    let passedArgs = paramNames.map { "\($0): \\(\($0))" }.joined()
```

Comment:
I guess you want `.joined(separator: ", ")` here, right?

Reply:
Yes, good catch